confgenerator/confgenerator_test.go

```diff
@@ -125,23 +127,23 @@ func TestGenerateConfsWithValidInput(t *testing.T) {
 	}
 }
 
-func expectedConfig(testName string, validFilePathFormat string, t *testing.T) string {
-	goldenPath := fmt.Sprintf(validFilePathFormat, platform, testName)
-	rawExpectedConfig, err := ioutil.ReadFile(goldenPath)
+func readFileContent(testName string, filePathFormat string, t *testing.T, respectGolden bool) []byte {
+	filePath := fmt.Sprintf(filePathFormat, platform, testName)
+	rawExpectedConfig, err := ioutil.ReadFile(filePath)
 	if err != nil {
-		if *updateGolden {
+		if *updateGolden && respectGolden {
 			// Tolerate the file not found error because we will overwrite it later anyway.
-			return ""
+			return []byte("")
 		} else {
-			t.Fatalf("test %q: error reading the golden conf from %s : %s", testName, goldenPath, err)
+			t.Fatalf("test %q: error reading the file from %s : %s", testName, filePath, err)
 		}
 	}
-	return string(rawExpectedConfig)
+	return rawExpectedConfig
 }
 
-func updateOrCompareGolden(t *testing.T, testName string, expected string, actual string, path string) {
+func updateOrCompareGolden(t *testing.T, testName string, expectedBytes []byte, actual string, path string) {
 	t.Helper()
-	expected = strings.ReplaceAll(expected, "\r\n", "\n")
+	expected := strings.ReplaceAll(string(expectedBytes), "\r\n", "\n")
```

Comment:
The reason https://travis-ci.com/github/GoogleCloudPlatform/ops-agent/jobs/500325327 failed is because those windows conf file, we never have the lines ending with `\r\n`. It's still `\n` as Unix..

We can use `unix2dos` to convert it to DOS
https://stackoverflow.com/questions/3569997/how-to-find-out-line-endings-in-a-text-file